tests/integration/vlan_test.py

```diff
@@ -90,6 +90,22 @@ def test_add_and_remove_two_vlans_on_same_iface(eth1_up):
     assert not current_state[Interface.KEY]
 
 
+def test_two_vlans_on_eth1_change_mtu(eth1_up):
+    desired_state = statelib.show_only(('eth1',))
+    desired_state[Interface.KEY].extend(TWO_VLANS_STATE[Interface.KEY])
+    for iface in desired_state[Interface.KEY]:
+        iface[Interface.MTU] = 2000
+    libnmstate.apply(desired_state)
+
+    del desired_state[Interface.KEY][1]
```

Comment:
could you do something a bit smarter than this, e.g. remove the ifstate based on name ? I'm looking at this and have no clue as to why it's the interface whose index is 1 that needs to be deleted.

Reply:
I'm recreating the situation raised in the ticket referenced in the commit message. If you take a look to TWO_VLANS_STATE, you would see that I am modifying eth1 and eth1.102 but not eth1.101.

If the workaround works, the MTU of the eth1.101 should be 2000 not 1500. I've also added another test in order to use a rollback scenario.

Reply:
You can also re-create the issue referenced in the ticket removing the ifstate by name, that won't hold you back :)

I think that it's OK if you delete ifstate[Interface.KEY][0], if there's only a single interface. It's easy to reason about that.

When there are multiple, it's trickier, order dependent, and, IMHO, confusing. As you say: I have to go and browse the TWO_VLANS_STATE to know which of the ifaces you're deleting. IMHO, that's not needed. 

I agree that checking TWO_VLANS_STATE for which interface has index 1 won't cost hundreds of millions of dollars over time, but ... Just saying.

Feel free to disregard, but readability is improved the way I'm suggesting.

Reply:
Oh sorry, I misunderstood you. You are completely right. Thanks!

Reply:
Done!